File: C++/A/143A-helpvasilisathewise2.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int r1, r2, c1, c2, d1, d2;
    cin >> r1 >> r2 >> c1 >> c2 >> d1 >> d2;
    unordered_set<int> nums;
    int a = (c1 - r2 + d1) / 2;
    if (a <= 9 && a > 0)
        nums.insert(a);

    int b = (d2 - c1 + r1) / 2;
    if (b <= 9 && b > 0)
        nums.insert(b);

    int c = (c1 - r1 + d2) / 2;
    if (c <= 9 && c > 0)
        nums.insert(c);

    int d = (d1 - c1 + r2) / 2;
    if (d <= 9 && d > 0)
        nums.insert(d);

    if (nums.size() == 4)
    {
        cout << a << ' ' << b << '\n'
             << c << ' ' << d << '\n';
    }
    else
        cout << -1;
}

File: Python/A/143A-helpvasilisathewise2.py
r1, r2 = map(int, input().split())
c1, c2 = map(int, input().split())
d1, d2 = map(int, input().split())

nums = list()
a = (c1 - r2 + d1) // 2
if (0 < a <= 9 and a not in nums):
    nums.append(a)
b = (d2 - c1 + r1) // 2
if (0 < b <= 9 and b not in nums):
    nums.append(b)
c = (c1 - r1 + d2) // 2
if (0 < c <= 9 and c not in nums):
    nums.append(c)
d = (d1 - c1 + r2) // 2
if (0 < d <= 9 and d not in nums):
    nums.append(d)
print(f"{nums[0]} {nums[1]}\n{nums[2]} {nums[3]}" if len(nums) == 4 else -1)
